### DISCUSSION TITLE
ScriptReference/GameObject-tag

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#25] [Notes]
# ScriptReference/GameObject-tag

https://docs.unity3d.com/ScriptReference/GameObject-tag.html

**Comment by vertxxyz** [comment]
Use [`CompareTag`](https://docs.unity3d.com/ScriptReference/GameObject.CompareTag.html) to compare tags instead of `==` or `Equals` as it's optimised for the purpose.
This property causes an allocation each time it is accessed, as the string is copied from the Unity native object into a managed C# string. `CompareTag` also prints an error if a tag does not exist, adding a layer of safety.
